### PR TITLE
Fix rebooting machines

### PR DESCRIPTION
**Pull request by kokesak**
As @mkluson suggested in the https://issues.redhat.com/browse/OAMG-6415 we can try add extra steps for reboot steps. The problem may be that the system reboots correctly, however the ansible play has trouble connecting back to the machine. 

This PR:

- use extra steps to the reboot role so it's able to reconnect to the machine
- there were 3 places where we had a reboot role implemented. Instead use just one role, so it's easier to maintain. This requires updating all test plans that use reboot.
- remove update_templates.yml file as it is not used anymore. We have an update role already.

### CODE REVIEW

**Comment by kokesak** [comment]
First round - no time outs

**Comment by kokesak** [comment]
It helps in a way that if the reboot timeouts, we still can reconnect to the machine and continue with a test:
```
08:37:04             out: TASK [Reboot into RHEL] ********************************************************
08:37:05             err: [WARNING]: sftp transfer mechanism failed on [root@10.29.163.143]. Use
08:37:05             err: ANSIBLE_DEBUG=1 to see detailed information
08:37:05             err: [WARNING]: scp transfer mechanism failed on [root@10.29.163.143]. Use
08:37:05             err: ANSIBLE_DEBUG=1 to see detailed information
10:39:53             out: fatal: [root@10.29.163.143]: FAILED! => {"changed": false, "elapsed": 7367, "msg": "Timed out waiting for last boot time check (timeout=1200)", "rebooted": true}
10:39:53             out: ...ignoring
10:39:53             out: 
10:39:53             out: TASK [Wait additional 15 minutes and try to reconnect (in case of previous task timeout/connection issue)] ***
10:39:54             err: [WARNING]: sftp transfer mechanism failed on [root@10.29.163.143]. Use
10:39:54             err: ANSIBLE_DEBUG=1 to see detailed information
10:39:54             err: [WARNING]: scp transfer mechanism failed on [root@10.29.163.143]. Use
10:39:54             err: ANSIBLE_DEBUG=1 to see detailed information
10:39:54             out: ok: [root@10.29.163.143]
10:39:54             out: 
10:39:54             out: TASK [Wait additional 5 minutes as last resort (really slow env)] **************
10:39:55             err: [WARNING]: sftp transfer mechanism failed on [root@10.29.163.143]. Use
10:39:55             err: ANSIBLE_DEBUG=1 to see detailed information
10:39:55             err: [WARNING]: scp transfer mechanism failed on [root@10.29.163.143]. Use
10:39:55             err: ANSIBLE_DEBUG=1 to see detailed information
10:39:55             out: ok: [root@10.29.163.143]
```
However, it can still take up to 2 hours to finish...

**Comment by mkluson** [comment]
> It helps in a way that if the reboot timeouts, we still can reconnect to the machine and continue with a test:
> 
> ```
> 08:37:04             out: TASK [Reboot into RHEL] ********************************************************
> 08:37:05             err: [WARNING]: sftp transfer mechanism failed on [root@10.29.163.143]. Use
> 08:37:05             err: ANSIBLE_DEBUG=1 to see detailed information
> 08:37:05             err: [WARNING]: scp transfer mechanism failed on [root@10.29.163.143]. Use
> 08:37:05             err: ANSIBLE_DEBUG=1 to see detailed information
> 10:39:53             out: fatal: [root@10.29.163.143]: FAILED! => {"changed": false, "elapsed": 7367, "msg": "Timed out waiting for last boot time check (timeout=1200)", "rebooted": true}
> 10:39:53             out: ...ignoring
> 10:39:53             out: 
> 10:39:53             out: TASK [Wait additional 15 minutes and try to reconnect (in case of previous task timeout/connection issue)] ***
> 10:39:54             err: [WARNING]: sftp transfer mechanism failed on [root@10.29.163.143]. Use
> 10:39:54             err: ANSIBLE_DEBUG=1 to see detailed information
> 10:39:54             err: [WARNING]: scp transfer mechanism failed on [root@10.29.163.143]. Use
> 10:39:54             err: ANSIBLE_DEBUG=1 to see detailed information
> 10:39:54             out: ok: [root@10.29.163.143]
> 10:39:54             out: 
> 10:39:54             out: TASK [Wait additional 5 minutes as last resort (really slow env)] **************
> 10:39:55             err: [WARNING]: sftp transfer mechanism failed on [root@10.29.163.143]. Use
> 10:39:55             err: ANSIBLE_DEBUG=1 to see detailed information
> 10:39:55             err: [WARNING]: scp transfer mechanism failed on [root@10.29.163.143]. Use
> 10:39:55             err: ANSIBLE_DEBUG=1 to see detailed information
> 10:39:55             out: ok: [root@10.29.163.143]
> ```
> 
> However, it can still take up to 2 hours to finish...

To solve this, we would need to request faster machines...

**Comment by kokesak** [comment]
Second run - no timeouts

**Comment by kokesak** [comment]
Ok, it seems that it fixes the timeout issue. I reworked this PR:

- use extra steps to the reboot role so it's able to reconnect to the machine
- there were 3 places where we had a reboot role implemented. Instead use just one role, so it's easier to maintain. This requires updating all test plans that use reboot.
- remove `update_templates.yml` file as it is not used anymore. We have an update role already.

**Comment by kokesak** [comment]
It seems that it fixes the reboot issue. The only "not-reliable" testing is on CentOS8.

File: tests/ansible_collections/roles/reboot/main.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
+- include: reboot.yml
```
Comment (mkluson) [inline tests/ansible_collections/roles/reboot/main.yml:2]: Is this file necessary? Could we use `tests/ansible_collections/roles/reboot/reboot.yml` directly instead?

File: tests/ansible_collections/roles/oracle-linux-specific/tasks/boot_standard_kernel.yml [hunk 2]
```diff
@@ -10,7 +10,3 @@
 
 - name: Set default kernel to Red Hat compatible kernel
   shell: "grubby --set-default /boot/vmlinuz-{{ kernel_ver.stdout }}"
-
-- name: Reboot
```
Comment (mkluson) [inline tests/ansible_collections/roles/oracle-linux-specific/tasks/boot_standard_kernel.yml:14]: How is this reboot replaced?

File: tests/ansible_collections/roles/reboot/main.yml (hunk 1 above)
Comment (kokesak) [reply tests/ansible_collections/roles/reboot/main.yml:2]: Yes no problem, we can add everything to just one file.

File: tests/ansible_collections/roles/oracle-linux-specific/tasks/boot_standard_kernel.yml (hunk 2 above)
Comment (kokesak) [reply tests/ansible_collections/roles/oracle-linux-specific/tasks/boot_standard_kernel.yml:14]: The reboot is done for everything after the "main preparation step" in `plans/main.fmf` file.

File: tests/ansible_collections/roles/reboot/main.yml (hunk 1 above)
Comment (kokesak) [reply tests/ansible_collections/roles/reboot/main.yml]: Changed.

**Review by danmyway — APPROVED** [review]
